packages/components/src/internal/components/entities/models.ts

```diff
@@ -555,34 +558,58 @@ export class OperationConfirmationData {
         } else {
             Object.assign(this, { notAllowed: [] });
         }
-        Object.assign(this, { idMap });
+        if (values?.notPermitted) {
+            values.notPermitted.forEach(npRow => {
+                idMap[caseInsensitive(npRow, 'rowId')] = false;
+            });
+        } else {
+            Object.assign(this, { notPermitted: [] });
+        }
+        Object.assign(this, {
+            idMap,
+            totalActionable: Object.values(idMap).filter(v => v === true).length,
+            totalNotActionable: Object.values(idMap).filter(v => v === false).length,
+        });
     }
 
-    isIdAllowed(id: number | string): boolean {
-        const idNum = typeof id === 'string' ? parseInt(id) : id;
+    isIdActionable(id: number | string): boolean {
+        const idNum = typeof id === 'string' ? parseInt(id, 10) : id;
         return this.idMap[idNum];
     }
 
-    get allAllowed(): boolean {
-        return this.notAllowed.length === 0;
+    getActionableIds(): number[] {
+        return this.allowed
+            .map(a => {
+                const rowId = a.RowId ? a.RowId : a;
+                if (!this.isIdActionable(rowId)) {
+                    return undefined;
+                } else {
+                    return rowId;
+                }
+            })
+            .filter(d => d !== undefined);
+    }
+
+    get allActionable(): boolean {
+        return this.totalNotActionable === 0;
     }
 
     // Note that this returns false if there are no samples represented since we generally want
     // noneAllowed to mean there are some samples but none are allowed.
```

Comment:
update `noneAllowed` in the comment as well?

Reply:
done